_posts/2020-12-14-changelog-137.markdown

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #137"
+date: 2020-12-14 17:00:00 +0300
+---
+
+## New Features
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6423 -->
+* Support `Find Usages` action (`Edit | Find Usages` or <kbd>Alt</kbd>+<kbd>F7</kbd>) for Cargo [features](https://doc.rust-lang.org/cargo/reference/features.html).
+Note, in order to make this feature work, you need to update your [TOML](https://plugins.jetbrains.com/plugin/8195-toml) plugin to the latest version as well
+
+  {% include gif-img.html path="/assets/posts/changelog-137/cargo-features-find-usages" w="700px" %}
+
+* [`cfg`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-attribute) attribute support improvements:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6436 -->
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6444 -->
+    * Provide completion for `feature` argument of `cfg` attribute
+      as well as completion for values of Cargo feature themselves
+
+      {% include gif-img.html path="/assets/posts/changelog-137/cfg-feature-completion" w="700px" %}
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6467 -->
+    * Gray out `cfg`-disabled items in structure view
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6432 -->
+    * Provide name resolution for Cargo features in [`doc(cfg())`](https://doc.rust-lang.org/beta/unstable-book/language-features/doc-cfg.html) attributes
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6466 -->
+    * Handle function arguments under `cfg` attribute during type inference and analyses
+
+
+* Support [`cfg_attr`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg_attr-attribute) attribute in some cases during name resolution and completion.
+  In general, the plugin now knows that `cfg_attr` can be expanded to other attributes, and the following stuff was adapted to use this knowledge:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6462 -->
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6460 -->
+    * Resolve and complete paths of `path` attribute and traits in `derive` attributes under `cfg_attr`
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6448 -->
+    * Take into account `cfg_attr` attribute during attribute and lint attribute completion
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6461 -->
+    * Resolve procedural macro [attributes](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) under `cfg_attr`
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6445 -->
+    * Add completion for conditional compilation options and operators inside `cfg_attr()` and `doc(cfg())` attributes
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6434 -->
+    * Provide name resolution for Cargo features in conditional attributes like `cfg_attr(docsrs, doc(cfg(feature = "macros")))`
```

Comment:
```suggestion
    * Provide name resolution for Cargo [features](https://doc.rust-lang.org/cargo/reference/features.html) in conditional attributes like `cfg_attr(docsrs, doc(cfg(feature = "macros")))`
```

Reply:
I'm not sure if we should add the hyperlink for "features" every time, but in this changelog, we already have two "Cargo features" with the link and three without it.

Reply:
We want to add link only for the first mention. I dropped the second hyperlink